### DISCUSSION TITLE
Innovations of velocity

### DISCUSSION THREAD

**Discussion by floosche2** [rodralez/NaveGo#41] [Q&A] [answered]
Dear NaveGo-Team, thank you for your very helpful work. 

I have a question concerning the innovations of the velocities zv.

In your paper you mention the measurement of the velocity as δyv = [vn − vn_G]^T (which is totally clear to me)

But in your code, you use:

> zv = (vel_e(i,:) - gps.vel(gdx,:) - ((omega_ie_n + omega_en_n) .* (DCMbn * gps.larm))' ...
            + (DCMbn * cross(wb_corrected, gps.larm) )' )';

Can you please tell me why you use different equations and give me a source of the equation you used in the code?

Thank you very much.

**Comment by ravindraji4** [comment]
Hi floosche2, 

the reference could be found in the paper titled "NaveGo: a simulation framework for low-cost integrated navigation systems" on page 114 and it is line under equation (25f). 

The link for this paper is on NaveGo **home page** Or directly click [http://ceai.srait.ro/index.php?journal=ceai&page=article&op=view&path%5B%5D=2478](Paper)

**Comment by rodralez** [comment, marked as answer]
> Dear NaveGo-Team, thank you for your very helpful work.
> 
> I have a question concerning the innovations of the velocities zv.
> 
> In your paper you mention the measurement of the velocity as δyv = [vn − vn_G]^T (which is totally clear to me)
> 
> But in your code, you use:
> 
> > zv = (vel_e(i,:) - gps.vel(gdx,:) - ((omega_ie_n + omega_en_n) .* (DCMbn * gps.larm))' ...
> > + (DCMbn * cross(wb_corrected, gps.larm) )' )';
> 
> Can you please tell me why you use different equations and give me a source of the equation you used in the code?

Hello, the extra terms in this equation are for correcting errors related to the lever arm between the IMU and the GNSS receiver. You can find this equation as Eq. 14.103 in Prof. Groves' book (2nd edition).  

Kind regards.
